Source/Core/Core/Scripting/Documentation/PythonRawAPIDocumentation.txt

```diff
@@ -0,0 +1,895 @@
+		  ConfigAPI.setBooleanConfigSettingForLayer("Base", "Main", "Interface", "debugModeEnabled", true)
+
+	void setEnumConfigSettingForLayer(String layerName, String systemName, String sectionName, String settingName, String enumTypeName, String enumValue):
+		- Sets the specified enum setting (of type enumTypeName) in the specified layer to the value specified by enumValue. If the layer does not contain that enum setting, then it is created.
+		  ConfigAPI.setEnumConfigSettingForLayer("GlobalGame", "Main", "Core", "SlotA", "EXIDeviceType", "MEMORYCARD")
+
+	void setFloatConfigSettingForLayer(String layerName, String systemName, String sectionName, String settingName, float newValue):
+		- Sets the specified float setting in the specified layer to newValue. If the layer does not contain that float setting, then it is created.
+		  ConfigAPI.setFloatConfigSettingForLayer("GlobalGame", "Main", "Core", "EmulationSpeed", 2.0)
+
+	void setSignedIntConfigSettingForLayer(String layerName, String systemName, String sectionName, String settingName, Integer newValue):
+		- Sets the specified signed int setting in the specified layer to newValue. If the layer does not contain that signed int setting, then it is created.
+		  ConfigAPI.setSignedIntConfigSettingForLayer("GlobalGame", "Main", "Core", "TimingVariance", 0)
+
+	void setStringConfigSettingForLayer(String layerName, String systemName, String sectionName, String settingName, String newValue):
+		- Sets the specified String setting in the specified layer to newValue. If the layer does not contain that String setting, then it is created.
+		  ConfigAPI.setStringConfigSettingForLayer("GlobalGame", "Main", "Core", "MemcardAPath", "dolphin/memory/memcardA")
+
+	void setUnsignedIntConfigSettingForLayer(String layerName, String systemName, String sectionName, String settingName, u32 newValue):
+		- Sets the specified unsigned int setting in the specified layer to newValue. If the layer does not contain that unsigned int setting, then it is created.
+		  ConfigAPI.setUnsignedIntConfigSettingForLayer("GlobalGame", "Main", "Core", "MEM1Size", 512456)
+
+	void saveSettings():
+		- Makes sure that the settings were saved to config files for the next session of Dolphin. Most settings are saved to config files as soon as they're updated, so this function might not really do anything.
+		  ConfigAPI.saveSettings()
+
+--------------------------------------------------------------------------------------------------------
+
+EmuAPI class:
+
+	void frameAdvance():
+		- This causes an error, since the frameAdvance() function is not supported for Python scripts. Instead, register a callback to run OnFrameStart, like in the example at the top of this documentation (you can then unregister the function when you want it to stop running - or you can call shutdownScript() to exit the script completely).
+		  EmuAPI.frameAdvance() ==> Throws an exception.
+
+	void loadState(String stateName):
```

Comment:
Is the Directory for these files accessible somehow? Such as in the Config API?

Reply:
Are you asking where EmuAPI is defined? If so, it's defined in Source/Core/Core/Scripting/InternalAPIModules/EmuAPI.h

Reply:
I'm talking about where the target Directory is. For example when I call `EmuAPI.saveState("myState.sav")`, can I get the fully qualified location it'll end up at, like `D:/Dolphin/SaveStates/myStates.sav`? Likewise when I call `EmuAPI.loadState("myState.sav")` is it checking that same Directory?

Reply:
I think the directory it defaults to is whatever the current working directory is. If you run the script in Visual Studio in debug mode, it will be in the Core/DolphinQT directory, whereas if you run it from the executable in your binary folder, then it will be in your binary folder (I think).

Reply:
Either way, saveState and loadState will both default to the same directory for a given run of Dolphin.